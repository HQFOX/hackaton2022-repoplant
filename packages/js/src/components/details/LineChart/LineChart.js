import React, { useEffect } from "react";
import { useTranslation } from "react-i18next";
import { HvLinechart } from "@hv/uikit-react-core";

const LineChart = ({ data, getLineData }) => {
  const { t } = useTranslation();

  useEffect(() => {
    getLineData();
  }, [getLineData]);

  if (!data?.length) return null;

  return (
    <HvLinechart
      title={t("components.details.chart.title")}
      subtitle={t("components.details.chart.subtitle")}
      data={data}
      xAxisTitle={t("components.details.chart.xAxisTitle")}
      yAxisTitle={t("components.details.chart.yAxisTitle")}
    />
  );
};

LineChart.propTypes = {};

export default LineChart;
