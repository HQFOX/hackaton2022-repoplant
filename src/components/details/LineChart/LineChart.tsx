import React, { useEffect } from "react";
import { useTranslation } from "react-i18next";
import { HvLinechart } from "@hv/uikit-react-core";

const LineChart = ({ data, getLineData }) => {
  const { t } = useTranslation();

  useEffect(() => {
    getLineData();
  }, [getLineData]);

  return data.length ? (
    <HvLinechart
      title={t("components.details.chart.title")}
      subtitle={t("components.details.chart.subtitle")}
      data={data}
      xAxisTitle={t("components.details.chart.xAxisTitle")}
      yAxisTitle={t("components.details.chart.yAxisTitle")}
    />
  ) : null;
};

export default LineChart;
