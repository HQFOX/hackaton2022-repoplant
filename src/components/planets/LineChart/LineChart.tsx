import React, { useEffect } from "react";
import { useTranslation } from "react-i18next";
import { HvLinechart } from "@hv/uikit-react-core";

const LineChart = ({ data, getLineData }) => {
  const { t } = useTranslation();

  useEffect(() => {
    getLineData();
  }, [getLineData]);

  return data.length ? (
    <HvLinechart
      title={t("components.planets.chart.title")}
      subtitle={t("components.planets.chart.subtitle")}
      data={data}
      xAxisTitle={t("components.planets.chart.xAxisTitle")}
      yAxisTitle={t("components.planets.chart.yAxisTitle")}
    />
  ) : null;
};

export default LineChart;
