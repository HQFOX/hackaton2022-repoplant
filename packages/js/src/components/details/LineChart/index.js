import { withStyles } from "@material-ui/core";
import styles from "./styles";
import LineChart from "./LineChart";

export default withStyles(styles, { name: "LineChart" })(LineChart);
