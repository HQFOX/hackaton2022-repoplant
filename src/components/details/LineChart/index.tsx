import { bindActionCreators, Dispatch } from "redux";
import { connect } from "react-redux";
import { getLineData } from "store/data/thunks";
import { withStyles } from "@material-ui/core";
import { AppState } from "typings/state";
import styles from "./styles";
import LineChart from "./LineChart";

const mapStateToProps = ({ data }: AppState) => ({
  data: data.line,
});

const mapDispatchToProps = (dispatch: Dispatch) =>
  bindActionCreators(
    {
      getLineData,
    },
    dispatch
  );

export default withStyles(styles, { name: "LineChart" })(
  connect(mapStateToProps, mapDispatchToProps)(LineChart)
);
