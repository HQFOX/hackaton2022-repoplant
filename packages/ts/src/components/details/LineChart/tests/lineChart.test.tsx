import React from "react";
import { mount } from "enzyme";
import LineChart from "..";

describe("<LineChart />", () => {
  let component;

  beforeEach(() => {
    component = mount(<LineChart />);
  });

  it("should be defined", () => {
    expect(component).toBeDefined();
  });

  it("matches the snapshot", () => {
    expect(component).toMatchSnapshot();
  });

  it("should be rendered", () => {
    const lineChart = component.find(LineChart);
    expect(lineChart.length).toBe(1);
  });
});
