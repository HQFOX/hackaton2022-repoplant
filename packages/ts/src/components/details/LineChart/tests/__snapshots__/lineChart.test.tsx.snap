// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LineChart /> matches the snapshot 1`] = `
<WithStyles(LineChart)>
  <LineChart
    classes={Object {}}
  />
</WithStyles(LineChart)>
`;
